### PR TITLE
added cargo audit and coverage to ci/cd

### PR DESCRIPTION
**Pull request by rwlockbg**

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Botanix Coverage
+
+on:
+    pull_request:
+    merge_group:
+        branches: [main, feature/poA-consensus]
+    push:
+        branches: [feature/poA-consensus]
```
Comment (arminsabouri) [inline .github/workflows/coverage.yml]: This branch is no longer used

**Review by arminsabouri — CHANGES_REQUESTED** [review]

**Comment by arminsabouri** [comment]
@rwlockbg just need to fix the conflict here

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (rwlockbg) [reply .github/workflows/coverage.yml]: removed

**Comment by arminsabouri** [comment]
@rwlockbg do you mind dropping the merge commits? In general i would rebase the latest changes rather than introducing merge commits.
https://github.com/botanix-labs/botanix/blob/master/CONTRIBUTING.MD

**Comment by arminsabouri** [comment]
Looks like we have some permission issues
```
debug - 2024-09-19 15:54:25,068 -- Commit creating result --- {"result": "RequestResult(error=RequestError(code='HTTP Error 403', params={}, description='{\"detail\":\"You do not have permission to perform this action.\"}'), warnings=[], status_code=403, text='{\"detail\":\"You do not have permission to perform this action.\"}')"}
error - 2024-09-19 15:54:25,0[68](https://github.com/botanix-labs/Macbeth/actions/runs/10944526500/job/30386605055?pr=361#step:9:69) -- Commit creating failed: {"detail":"You do not have permission to perform this action."}
```

**Comment by rwlockbg** [comment]
> Looks like we have some permission issues
> 
> ```
> debug - 2024-09-19 15:54:25,068 -- Commit creating result --- {"result": "RequestResult(error=RequestError(code='HTTP Error 403', params={}, description='{\"detail\":\"You do not have permission to perform this action.\"}'), warnings=[], status_code=403, text='{\"detail\":\"You do not have permission to perform this action.\"}')"}
> error - 2024-09-19 15:54:25,0[68](https://github.com/botanix-labs/Macbeth/actions/runs/10944526500/job/30386605055?pr=361#step:9:69) -- Commit creating failed: {"detail":"You do not have permission to perform this action."}
> ```

@birozuru SInce I am not the owner of the repo, this error pops up. Rememeber that when you run the coverage, it was all fine with codecov ?

**Comment by birozuru** [comment]
> > Looks like we have some permission issues
> > ```
> > debug - 2024-09-19 15:54:25,068 -- Commit creating result --- {"result": "RequestResult(error=RequestError(code='HTTP Error 403', params={}, description='{\"detail\":\"You do not have permission to perform this action.\"}'), warnings=[], status_code=403, text='{\"detail\":\"You do not have permission to perform this action.\"}')"}
> > error - 2024-09-19 15:54:25,0[68](https://github.com/botanix-labs/Macbeth/actions/runs/10944526500/job/30386605055?pr=361#step:9:69) -- Commit creating failed: {"detail":"You do not have permission to perform this action."}
> > ```
> 
> @birozuru SInce I am not the owner of the repo, this error pops up. Rememeber that when you run the coverage, it was all fine with codecov ?

Yeah seems like it's permission issues on your github_token

**Comment by rwlockbg** [comment]
@birozuru @0xBEEFCAF3 lets merge this MR and @birozuru will replace my token with his codedev token ?

**Comment by birozuru** [comment]
> @birozuru @0xBEEFCAF3 lets merge this MR and @birozuru will replace my token with his codedev token ?

Gonna replace now and rerun the workflow.

**Review by birozuru — APPROVED** [review]
Codecov token updated. LGTM

**Review by arminsabouri — APPROVED** [review]

**Pull request merged**